commands/core/config.drush.inc

```diff
@@ -367,12 +371,30 @@ function _drush_config_export($destination, $destination_dir, $branch) {
   if (!$comment) {
     $comment = "Exported configuration.";
   }
+  $storage_filters = _drush_config_get_storage_filters();
   if (count(glob($destination_dir . '/*')) > 0) {
     // Retrieve a list of differences between the active and target configuration (if any).
     $target_storage = new FileStorage($destination_dir);
     /** @var \Drupal\Core\Config\StorageInterface $active_storage */
     $active_storage = Drupal::service('config.storage');
-    $config_comparer = new StorageComparer($active_storage, $target_storage, Drupal::service('config.manager'));
+    $comparison_source = $active_storage;
+
+    // If the output is being filtered, then write a temporary copy before doing
+    // any comparison.
+    if (!empty($storage_filters)) {
+      $tmpdir = drush_tempdir();
+      drush_copy_dir($destination_dir, $tmpdir, FILE_EXISTS_OVERWRITE);
+      $comparison_source = new FileStorage($tmpdir);
+      $comparison_source_filtered = new StorageWrapper($comparison_source, $storage_filters);
+      foreach ($active_storage->listAll() as $name) {
+        // Copy active storage to our temporary active store.
+        if ($existing = $active_storage->read($name)) {
+          $comparison_source_filtered->write($name, $existing);
+        }
+      }
+    }
```

Comment:
Yeah, Drush just instantiates its own FileStorage objects.  Drupal could in theory add filters on top of the objects the client provided, but we would need to be able to hook into this process somehow if we wanted to be able to do a diff and tell the user what was about to change.